text/0002-modules.md

```diff
@@ -0,0 +1,58 @@
+# Modules
+- Start Date: 2019/04/29
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/2
+
+## Decision, including impact on distributions
+In-browser Javascript modules are the OpenMRS mechanism for adding functionality and creating frontend code for distributions.
```

Comment:
Would this be the equivalent of what we currently have where an end user goes to our [module repository](https://addons.openmrs.org), downloads a module or open web app, and clicks a button to install it at runtime to extend the the user interface? https://wiki.openmrs.org/x/QQEr
https://wiki.openmrs.org/x/RAEr

Reply:
👍 yes

Both developers and end users could have the ability to pick and choose which frontend javascript modules to include in their application. Developers of distributions would definitely always have this choice, and for end users we would have to write some code and maybe a UI to help them do this, but it is 100% possible with javascipt modules. Sounds like something that we should probably build for end users.